generate/util/util.go

```diff
@@ -0,0 +1,235 @@
+package util
```

Comment:
How about naming the package `genutil`? Seems like it could communicate a little better what it's intended to be used for?

Could be a good idea to add a `doc.go` with a package documentation stating that the functions are here to support generated code and should not be accessed directly.